app/controllers/newsletter_signups_controller.rb

```diff
@@ -0,0 +1,30 @@
+class NewsletterSignupsController < ApplicationController
+  before_action :set_newsletter_signup
+
+  def create
+    response = request_signup
```

Comment:
I'm not sure I have my mind totally wrapped around the error handling here but I gave it a shot. Looking forward to your suggestions.